python3/packages/observer.py

```diff
@@ -288,12 +289,13 @@ def instrument_function(wrapped, _, args, kwargs):
                     # class or classmethod
                     aspan.set_attribute("xs.span.args.str", str(args))
                     aspan.set_attribute("xs.span.kwargs.str", str(kwargs))
-                else:
-                    # function, staticmethod or instancemethod
+                elif type(wrapped)==types.FunctionType or type(wrapped)==types.MethodType:
                     bound_args = inspect.signature(wrapped).bind(*args, **kwargs)
                     bound_args.apply_defaults()
                     for k, v in bound_args.arguments.items():
                         aspan.set_attribute(f"xs.span.arg.{k}", str(v))
+                else: #eg. class 'PartialCallableObjectProxy'
```

Comment:
if it's a class, shouldn't this have been caught in the if at line 288?

Reply:
Yes indeed. And, as the `else` is followed by a `pass`, it should be superfluous. I added a suggestion to remove it below.

Reply:
I meant class in the same sense as Function and Method are classes.